homeassistant/components/websocket_api/messages.py

```diff
@@ -65,12 +65,31 @@ def construct_result_message(iden: int, payload: str) -> str:
     return f'{{"id":{iden},"type":"result","success":true,"result":{payload}}}'
 
 
-def error_message(iden: int | None, code: str, message: str) -> dict[str, Any]:
+def error_message(
+    iden: int | None,
+    code: str,
+    message: str,
+    translation_key: str | None = None,
+    translation_domain: str | None = None,
+    translation_placeholders: dict[str, Any] | None = None,
+) -> dict[str, Any]:
     """Return an error result message."""
+    error_payload: dict[str, Any] = {
+        "code": code,
+        "message": message,
+    }
+    # In case a ServiceNotFound error is raised
+    # we use 'websocket_api' as default translation_domain.
+    # In case `translation_key` is `None` we do not set it, nor the
+    # `translation`_placeholders` and `translation_domain`.
+    if translation_key is not None:
+        error_payload["translation_key"] = translation_key
+        error_payload["translation_placeholders"] = translation_placeholders or {}
```

Comment:
Frontend should allow this to be null or a map
```suggestion
        error_payload["translation_placeholders"] = translation_placeholders
```